### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by michelg10**
Created onboarding page and `MarkdownText`, `ProgressBar`, and `ToggleGridButtonsComponent` components

### CODE REVIEW

File: texas-hearing-institute/pages/Onboarding1.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    // This effect triggers every time 'itemsSelected' changes, storing the new state
+    useEffect(() => {
+        storeItemSelection(storageKey, consonants, itemsSelected);
+    }, [itemsSelected]); // Re-run this effect when 'itemsSelected' changes
+
+
+
+    return (
+        <SafeAreaView style={{
```
Comment (pathaks1) [inline texas-hearing-institute/pages/Onboarding1.tsx:68]: SafeAreaView is iOS specific - talk with Paul and Vincent for how they made it Android safe too with React Navigation

File: texas-hearing-institute/pages/Onboarding2.tsx
Comment (pathaks1) [inline texas-hearing-institute/pages/Onboarding2.tsx:1]: same comment on safeareaview for this and rest of onboarding pages; might be nice to take functions like storeItemSelection, retrieveItemSelections, etc and put them into a single function file and use those

Comment (pathaks1) [inline texas-hearing-institute/pages/Onboarding2.tsx:1]: same comment on safeareaview for this and rest of onboarding pages; might be nice to take functions like storeItemSelection, retrieveItemSelections, etc and put them into a single function file and use those

File: texas-hearing-institute/pages/Onboarding1.tsx (hunk 1 above)
Comment (pathaks1) [inline texas-hearing-institute/pages/Onboarding1.tsx]: SafeAreaView is iOS specific - talk with Paul and Vincent for how they made it Android safe too with React Navigation

**Review by pathaks1 — CHANGES_REQUESTED** [review]

File: texas-hearing-institute/App.tsx [hunk 2]
```diff
@@ -1,21 +1,103 @@
...
+            return itemsSelected;
+
+        } catch (e) {
+            console.log(e);
+            return items.map(() => false);
+        }
+    }
+
+    // Load stored selection statuses when the component mounts
+    useEffect(() => {
```
Comment (pathaks1) [inline texas-hearing-institute/App.tsx]: useEffect should be used for things like side effects; I think it is better design to load in the components in useState itself; please try making this change throughout all the files (I have sent you a screenshot on Slack with an example of how to use async storage within a useState)

**Review by pathaks1 — CHANGES_REQUESTED** [review]

File: texas-hearing-institute/components/CustomSafeAreaView/CustomSafeAreaView.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { View } from 'react-native';
+import React, { ReactNode } from 'react';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+interface CustomSafeAreaViewType {
+	children: ReactNode;
+}
+
+const CustomSafeAreaView: React.FC<CustomSafeAreaViewType> = ({ children }) => {
```
Comment (baylee-rl) [inline texas-hearing-institute/components/CustomSafeAreaView/CustomSafeAreaView.tsx:9]: it's better style to leave out the React.FC typing for functional components

File: texas-hearing-institute/pages/Onboarding/OnboardingPage.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+import {
+	setupPrompts,
+	setupPersistenceKeys,
+	setupPageElements,
+} from '../../utils/soundInventoryDataAndKeys';
+import CustomSafeAreaView from '../../components/CustomSafeAreaView/CustomSafeAreaView';
+
+// commented out typing, teporarily said to be 'any' in parameters
+
+// type RootStackParamList = {
```
Comment (baylee-rl) [inline texas-hearing-institute/pages/Onboarding/OnboardingPage.tsx:21]: are u going to use all this commented out stuff later? lol

**Review by baylee-rl — APPROVED** [review]
looks good! going to merge now. keep working on new branch

**Pull request merged**